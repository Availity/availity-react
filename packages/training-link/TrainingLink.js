import React from 'react';
import PropTypes from 'prop-types';

const shouldProbablyBeInUIKit = {
  paddingTop: '12px',
};

const TrainingLink = ({ name, link }) => (
  <span style={shouldProbablyBeInUIKit} className="ml-auto">
    Need Help?{' '}
    <a href={link} target="_blank" rel="noopener noreferrer">
      Watch a demo
    </a>{' '}
    for {name}
  </span>
);

TrainingLink.propTypes = {
  link: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
};

export default TrainingLink;
