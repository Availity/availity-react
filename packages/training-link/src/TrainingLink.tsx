import React from 'react';

const shouldProbablyBeInUIKit = { paddingTop: '12px' };
const linkStyles = { fontWeight: 'bold' } as const;

export type TrainingLinkProps = {
  name: string;
  link: string;
};

const TrainingLink = ({ name, link }: TrainingLinkProps): JSX.Element => (
  <span style={shouldProbablyBeInUIKit} className="ml-auto">
    Need Help?{' '}
    <a href={link} style={linkStyles} className="link" target="_blank" rel="noopener noreferrer">
      Watch a demo
    </a>{' '}
    <span className="sr-only">(opens in a new tab) </span>
    for {name}
  </span>
);

export default TrainingLink;
