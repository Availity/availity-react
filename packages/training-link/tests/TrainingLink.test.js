import React from 'react';
import { render, cleanup } from '@testing-library/react';
import TrainingLink from '..';

afterEach(cleanup);

describe('TrainingLink', () => {
  test('should not render with link and name', () => {
    const { container } = render(
      <TrainingLink link="http://catvidoes.com" name="Appeals" />
    );

    expect(container).toMatchSnapshot();
  });
});
