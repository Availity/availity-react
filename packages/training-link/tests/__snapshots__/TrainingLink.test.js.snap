// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrainingLink should not render with link and name 1`] = `
<div>
  <span
    class="ml-auto"
    style="padding-top: 12px;"
  >
    Need Help?
     
    <a
      href="http://catvidoes.com"
      rel="noopener noreferrer"
      target="_blank"
    >
      Watch a demo
    </a>
     
    for 
    Appeals
  </span>
</div>
`;
